python_modules/libraries/dagster-aws/dagster_aws/ecs/launcher.py

```diff
@@ -239,19 +240,42 @@ def _task_definition(self, metadata, image):
             task_definition = self.ecs.describe_task_definition(taskDefinition=self.task_definition)
             return task_definition["taskDefinition"]
 
+        secrets = merge_dicts(
+            (
+                get_tagged_secrets(self.secrets_manager, self.secrets_tag)
+                if self.secrets_tag
+                else {}
+            ),
+            get_secrets_from_arns(self.secrets_manager, self.secrets),
+        )
+        secrets_dict = (
+            {"secrets": [{"name": key, "valueFrom": value} for key, value in secrets.items()]}
+            if secrets
+            else {}
+        )
+
+        task_definition = {}
+        with suppress(ClientError):
+            # TODO: Vary family name by repository location
+            task_definition = self.ecs.describe_task_definition(taskDefinition="dagster-run")[
+                "taskDefinition"
+            ]
```

Comment:
So i'm imagining a situation where there's a big run queue that's way behind. 50 runs get queued up for location X using image A. location X updates its image to image B. 50 more runs get queued up. They're prioritized in the queue such a way that the run launcher is switching back and forth between the two images every time it launches a run.

That is admittedly pretty unlikely/pathological, but is that an example of the kind of re-registration cycle that you're worried about?

One fun thing that you could do is create some kind of a snapshot / hash of the dictionary that uniquely defines the run - and append that hash to the family name. We used to do something kind of like that in cloud to be able to identify a specific pod name for a specific set of metadata for a location. That may be overkill though.

The important thing is that two of these definitions still be able to co-exist even when they're part of the same family.